osu.Game/Overlays/ModSelection/ModButton.cs

```diff
@@ -0,0 +1,80 @@
+﻿//Copyright (c) 2007-2016 ppy Pty Ltd <contact@ppy.sh>.
+//Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using OpenTK.Graphics;
+using osu.Framework;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Primitives;
+using osu.Framework.Input;
+using osu.Game.Modes;
+using osu.Game.Modes.UI;
+
+namespace osu.Game.Overlays.ModSelection
+{
+    class ModButton : ClickableContainer, IStateful<ModButtonState>
+    {
+        private Mod Mod;
+        private Color4 bgColor;
+
+        public ModButtonState State{get; set;}
+
+        private const int transform_time = 150;
+
+        public ModButton(Mod mod, Color4 color, bool isActive)
+        {
+            Mod = mod;
+            bgColor = color;
+
+            State = ModButtonState.Disabled;
+
+            RelativeSizeAxes = Axes.Y;
+            Width = 100;
+
+            Origin = Anchor.Centre;
+            Anchor = Anchor.CentreLeft;
+
+            Margin = new MarginPadding { Left = 40, Top = -15 };
+            Children = new Drawable[]
+            {
+                new DrawableMod
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Origin = Anchor.Centre,
+                    Anchor = Anchor.Centre,
+                    Mod = Mod,
+                    Colour = bgColor
+                }
+            };
+
+            if (isActive)
+                Arm();
+        }
+        protected override bool OnClick(InputState state)
+        {
+            if (State == ModButtonState.Disabled)
+                Arm();
+            else
+                Disarm();
+
+            return base.OnClick(state);
+        }
+        private void Arm()

```

Comment:
private methods should be lower case. but why even bother? bind to OnStateChange and apply the transforms, then just set the state rather than calling Arm/Disarm.

I'd also use Enabled/Disabled, not Armed